src/games/seabattle/game.ts

```diff
@@ -116,12 +118,14 @@ export const SeabattleGame = Game({
 // Helper function for generating random ships positioning.
 export function generateRandomShips(player: number): IShip[] {
   let result: IShip[];
+  let shipID;
   do {
     result = [];
     for (const shipSize of VALID_SHIPS_SIZES) {
       const count: number = VALID_SHIPS_COUNT[shipSize];
       for (let i = 0; i < count; i++) {
-        result.push(randomlyGetShip(player, shipSize));
+        shipID = shortid.generate();
+        result.push(randomlyGetShip(player, shipSize, shipID));
```

Comment:
We should avoid setShips being non deterministic given that we might need to replay the game. Please just check that they are unique and set to whatever is provided. Generate the random strings on the UI code.

Reply:
discussed on gho